platform/security/security-oidc-bundle/pom.xml

```diff
@@ -0,0 +1,196 @@
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <enforceBytecodeVersion>
+                                    <maxJdkVersion>1.8</maxJdkVersion>
+                                </enforceBytecodeVersion>
+                            </rules>
+                            <fail>true</fail>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>check-artifact-size</id>
+                        <phase>verify</phase>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <ArtifactSizeEnforcerRule
+                                  implementation="org.codice.maven.artifactsize.ArtifactSizeEnforcerRule">
+                                    <maxArtifactSize>2.0_MB</maxArtifactSize>
+                                </ArtifactSizeEnforcerRule>
+                            </rules>
+                        </configuration>
+                    </execution>
+                </executions>
+                <dependencies>
+                    <dependency>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>extra-enforcer-rules</artifactId>
+                        <version>1.1</version>
+                    </dependency>
+                </dependencies>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Embed-Dependency>
+                            jbcrypt,
+                            pac4j-core,
+                            pac4j-oauth,
+                            pac4j-oidc,
+                            pac4j-jwt,
+                            siv-mode
+                        </Embed-Dependency>
+                        <Conditional-Package>org.codice.ddf.security.util</Conditional-Package>
+                        <Import-Package>
+                            !com.github.scribejava.*,
+                            !com.google.common.*,
+                            !org.bouncycastle.*,
+                            !org.joda.time,
+                            !org.pac4j.*,
+                            com.google.crypto.tink.*;resolution:=optional,
+                            net.shibboleth.utilities.java.support.*;resolution:=optional,
+                            org.apache.shiro.*;resolution:=optional,
+                            org.opensaml.*;resolution:=optional,
+                            org.springframework.security.crypto.password;resolution:=optional,
```

Comment:
This goes for all of the optionals here they should be excluded if they aren't actually optional. Pretty sure their is a hidden cost to optional in karaf it will cause this bundle and anything that depends on it to be refreshed on feature install even when its not necessary so it can cause startup issues.

Reply:
We can take a look at making these excluded instead.

Reply:
These optional imports at one point corresponded to optional dependencies of the embedded artifacts. The embed list has changed, though, so it may be possible to exclude some of them now.